kernel/input_test.go

```diff
@@ -275,3 +277,285 @@ func TestInputKernelConverge(t *testing.T) {
 	testfunc(t, k00)
 	testfunc(t, k10)
 }
+
+func TestAssignable(t *testing.T) {
+	ctx := cuecontext.New()
+
+	var blah string
+	tt := map[string]struct {
+		T       interface{}
+		cue     string
+		invalid bool
+	}{
+		"nonpointer": {
```

Comment:
@papagian @kylebrandt @radiohead if you have 15 minutes to look over this list of relations and suggest/ask about any you don't see represented, that'd be lovely. The current set could be described as "decent, but not even pretending to be exhaustive."